artemis-protocols/artemis-mqtt-protocol/src/main/java/org/apache/activemq/artemis/core/protocol/mqtt/MQTTSession.java

```diff
@@ -30,10 +30,12 @@
 
 public class MQTTSession {
 
-   static Map<String, MQTTSessionState> SESSIONS = new ConcurrentHashMap<>();
+   private static Map<String, MQTTSessionState> SESSIONS = new ConcurrentHashMap<>();
```

Comment:
Looking at this im bit worried that this is a static in first place, e.g. its possible to have multiple embedded brokers, in that case each broker shouldnt share any state, by being static means they would.

Reply:
i sent a possible solution to address this, in a pr to your branch.